### PR TITLE
[ADD] Added web_widget_digital_signature module

### PR DESCRIPTION
**Pull request by JayVora-SerpentCS**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please call it _web_widget_digital_signature_ for following current conventions and for clarity.

File: web_digital_sign/__init__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import users
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (pedrobaeza) [inline web_digital_sign/__init__.py]: vim line not needed.

File: web_digital_sign/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+{
+    "name" : "Web Digital Signature",
+    "version" : "1.0",
+    "author" : "Serpent Consulting Services Pvt. Ltd.",
+    "category": '',
+    'complexity': "easy",
+    'depends': ['web'],
+    "description": """
+        This module provides the functionality to store digital signature image for a record.
```
Comment (pedrobaeza) [inline web_digital_sign/__openerp__.py]: You can use OCA's description template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: web_digital_sign/static/src/css/jquery-ui.css [hunk 3]
```diff
@@ -0,0 +1,1225 @@
+/*! jQuery UI - v1.11.2 - 2014-10-16
```
Comment (pedrobaeza) [inline web_digital_sign/static/src/css/jquery-ui.css]: Why do you need this? Isn't jQuery already bundled with web client?

Anyways, this is too heavy for only a couple of styles you surely need, don't you think?

**Comment by pedrobaeza** [comment]
Please remove company logo, because OCA modules shouldn't have it.

File: web_digital_sign/users.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#    
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2004-2010 OpenERP SA (<http://www.openerp.com>)
+#    Copyright (C) 2011-2015 Serpent Consulting Services Pvt. Ltd. (<http://www.serpentcs.com>).
```
Comment (pedrobaeza) [inline web_digital_sign/users.py]: Please respect 80 cols limit

File: web_digital_sign/users.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (pedrobaeza) [inline web_digital_sign/users.py]: Put it under 'models' folder and call it res_users.py for following conventions.

File: web_digital_sign/users.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+from openerp.osv import osv, fields
+
+class Users(osv.Model):
+    _inherit = 'res.users'
+    
+    _columns = {
+        'signature_image': fields.binary('Signature')
+    }
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
Comment (pedrobaeza) [inline web_digital_sign/users.py]: Remove vim line

File: web_digital_sign/users.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import osv, fields
+
+class Users(osv.Model):
```
Comment (pedrobaeza) [inline web_digital_sign/users.py]: Use orm.Model

File: web_digital_sign/users_view.xml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+<openerp>
+	<data>
+
+		<record id="inherited_res_users_form" model="ir.ui.view">
+            <field name="name">inherited.res.users.form</field>
+            <field name="model">res.users</field>
+            <field name="inherit_id" ref="base.view_users_form"/>
+            <field name="arch" type="xml">
+            	<xpath expr="//field[@name='signature']" position="replace">
+					<label for="signature_image" class="oe_edit_only"/>
```
Comment (pedrobaeza) [inline web_digital_sign/users_view.xml]: Don't mix tabs and spaces as indentator.

**Comment by gurneyalex** [comment]
@JayVora-SerpentCS I've linked your github account to the OCA database, but I could not find a CLA for your account.

**Comment by JayVora-SerpentCS** [comment]
I am sorry If I missed, but we are regtd .https://github.com/odoo/odoo/blob/8.0/doc/cla/corporate/SerpentCS.md
Did you mean we have to sign CLA different than this one?
Thanks.

**Comment by legalsylvain** [comment]
Hi, 
There are some flake8 & pylint bugs to fix.
(https://travis-ci.org/OCA/web/builds/56562389)

File: web_digital_sign/static/src/css/jquery-ui.css (hunk 3 above)
Comment (legalsylvain) [inline web_digital_sign/static/src/css/jquery-ui.css]: agree with @pedrobaeza here.

**Comment by hbrunn** [comment]
I put a couple of comments in https://github.com/OCA/web/pull/115, especially the javascript ones apply to this PR just the same

File: web_digital_sign/__init__.py (hunk 1 above)
Comment (JayVora-SerpentCS) [inline web_digital_sign/__init__.py]: Fixed.

File: web_digital_sign/__openerp__.py (hunk 2 above)
Comment (JayVora-SerpentCS) [inline web_digital_sign/__openerp__.py]: Added description as per template.

File: web_digital_sign/users.py (hunk 4 above)
Comment (JayVora-SerpentCS) [inline web_digital_sign/users.py]: Fixed.

File: web_digital_sign/users.py (hunk 5 above)
Comment (JayVora-SerpentCS) [inline web_digital_sign/users.py]: Done.

File: web_digital_sign/users.py (hunk 6 above)
Comment (JayVora-SerpentCS) [inline web_digital_sign/users.py]: Fixed.

File: web_digital_sign/users.py (hunk 7 above)
Comment (JayVora-SerpentCS) [inline web_digital_sign/users.py]: Fixed.

**Comment by StefanRijnhart** [comment]
I think pylint wants you to use relative imports (from . import res_users) and flake8 found

 **openerp**.py:45:11: E126 continuation line over-indented for hanging indent

**Comment by JayVora-SerpentCS** [comment]
Goes All well!
Thanks OCA!

File: web_widget_digital_signature/README.rst [hunk 9]
```diff
@@ -0,0 +1,2 @@
+This module provides the functionality to store digital signature image for a record.
```
Comment (OSguard) [inline web_widget_digital_signature/README.rst]: please use template:
https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: web_widget_digital_signature/__openerp__.py [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    "name": "Web Widget Digital Signature",
+    "version": "1.0",
+    "author": "Serpent Consulting Services Pvt. Ltd.",
```
Comment (OSguard) [inline web_widget_digital_signature/__openerp__.py]: add ",Odoo Community Association (OCA)" as coauthor

File: web_widget_digital_signature/__openerp__.py [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    "category": '',
+    'depends': ['web'],
+    "description": """
```
Comment (OSguard) [inline web_widget_digital_signature/__openerp__.py:31]: you can remove description if you have a readme.rst

Comment (JayVora-SerpentCS) [inline web_widget_digital_signature/__openerp__.py:31]: Managed to do those changes!
On Tuesday 02 June 2015 03:46 PM, Markus Schneider wrote:

> In web_widget_digital_signature/**openerp**.py 
> https://github.com/OCA/web/pull/105#discussion_r31508953:
> 
> > +#    but WITHOUT ANY WARRANTY; without even the implied warranty of
> > +#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> > +#    GNU Affero General Public License for more details.
> > +#
> > +#    You should have received a copy of the GNU Affero General Public License
> > +#    along with this program.  If not, see http://www.gnu.org/licenses/.
> > +#
> > +##############################################################################
> > +
> > +{
> > -    "name": "Web Widget Digital Signature",
> > -    "version": "1.0",
> > -    "author": "Serpent Consulting Services Pvt. Ltd.",
> > -    "category": '',
> > -    'depends': ['web'],
> > -    "description": """
> 
> you can remove description if you have a readme.rst
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/OCA/web/pull/105/files#r31508953.

## 

Thanks,
Regards,

Serpent Consulting Services.
Web : http://www.serpentcs.com
Twitter: jaynvora, husendaudi
Skype: jaynvora, husen.daudi
Blog : http://www.serpentcs.com/blog

Comment (bwrsandman) [inline web_widget_digital_signature/__openerp__.py:31]: Actually, that is not true in v7. You still need it here.

File: web_widget_digital_signature/models/res_users.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+from openerp.osv import orm, fields
+
+
+class Users(orm.Model):
+    _inherit = 'res.users'
+
+    _columns = {
+        'signature_image': fields.binary('Signature'),
+        'signature_image2': fields.binary('sSignature'),
```
Comment (bwrsandman) [inline web_widget_digital_signature/models/res_users.py]: What is sSignature?

Comment (ghost) [inline web_widget_digital_signature/models/res_users.py]: It's just a field name which states that it's a second signature.

Comment (bwrsandman) [inline web_widget_digital_signature/models/res_users.py]: This could be improved by using a more descriptive name or string or help text.

**Comment by JayVora-SerpentCS** [comment]
We are good to go!
Do inform us if anything remains.
Thanks.

**Comment by OSguard** [comment]
Runbot not loading, need to rebase to fix this bug!

**Comment by bwrsandman** [comment]
Add `&debug` to the end of the runbot URL and open js examiner you will see that there's a JS error:

09:51:18,864 TypeError: moment.tz is undefined1 web_pytz.js:55:12

**Comment by OSguard** [comment]
@bwrsandman this is a bug we already fix in OCA/7.0 3 weeks ago, this branch needs only to rebase!

**Comment by bwrsandman** [comment]
Ah yes... because runbot doesn't merge https://github.com/OCA/runbot-addons/issues/45

**Comment by ghost** [comment]
@oca

We would be glad to know if this can be merged quickly.

Thanks.

**Comment by StefanRijnhart** [comment]
@MeetSCS you were asked to rebase some time ago. Please do that first, and then the review can continue.

**Comment by ghost** [comment]
Branch is rebased.

**Comment by dreispt** [comment]
There's something wrong with the PR: the diff shows files not belonging to the proposed module. :bomb:

**Comment by JayVora-SerpentCS** [comment]
Do we have any blocker?

**Comment by pedrobaeza** [comment]
Yeah, please see last comment about the PR with commits that are not valid.

**Comment by pedrobaeza** [comment]
Is this going to be continued? In v8 it was merged with other name: https://github.com/OCA/web/tree/8.0/web_widget_digitized_signature

**Comment by StefanRijnhart** [comment]
Well, last comment was 13 months ago. Time to close.

**Pull request closed without merging**